Modules/_functoolsmodule.c

```diff
@@ -1175,6 +1184,11 @@ lru_cache_new(PyTypeObject *type, PyObject *args, PyObject *kw)
         return NULL;
     }
 
+    if (make_key != Py_None && !PyCallable_Check(make_key)) {
+        PyErr_SetString(PyExc_TypeError, "the make_key argument must be callable");
+        return NULL; 
+    }
+
```

Comment:
If `typed` is passed and make_key exists, flow should be identical to the .py implementation.